.github/workflows/continuous_integration.yml

```
@@ -19,10 +19,10 @@ jobs:
           pipenv install pre_commit
           pipenv run python -m pre_commit run --all-files
   test:
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-22.04
```

```suggestion
    runs-on: ubuntu-24.04
```
I guess not 24.04 to have older python?

---

older libraries overall, to make sure we are backwards compatible